### PR TITLE
make seech service extend base service and use the auth helper

### PR DESCRIPTION
**Pull request by amsacha**
### Motivation and Context
Make SpeechService extend the BaseService class and use the AuthHelper class. This also fixes a bug where the speech service did not support the ApiKey auth.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: samples/apps/copilot-chat-app/webapp/src/components/chat/ChatInput.tsx [hunk 1]
```diff
@@ -79,15 +79,15 @@ export const ChatInput: React.FC<ChatInputProps> = (props) => {
         async function initSpeechRecognizer() {
             const speechService = new SpeechService(process.env.REACT_APP_BACKEND_URI as string);
 
-            var response = await speechService.validSpeechKeyAsync();
+            var response = await speechService.validSpeechKeyAsync(await AuthHelper.getSKaaSAccessToken(instance, inProgress));
```
Comment (hathind-ms) [inline samples/apps/copilot-chat-app/webapp/src/components/chat/ChatInput.tsx:82]: not related to this PR, rename to "validateSpeechKeyAsync"

Comment (amsacha) [reply samples/apps/copilot-chat-app/webapp/src/components/chat/ChatInput.tsx]: done

**Review by glahaye — DISMISSED** [review]

File: samples/apps/copilot-chat-app/webapp/src/libs/services/SpeechService.ts [hunk 2]
```diff
@@ -44,39 +38,15 @@ export class SpeechService {
...
-            const requestUrl = new URL(commandPath, this.serviceUrl);
-            const response = await fetch(requestUrl, {
-                method: method ?? 'GET',
-                headers: { 'Content-Type': 'application/json' },
-            });
-
-            if (!response.ok) {
-                throw Object.assign(new Error(response.statusText + ' => ' + (await response.text())));
-            }
+    private invokeTokenAsync = async (accessToken: string): Promise<TokenResponse> => {
```
Comment (teresaqhoang) [inline samples/apps/copilot-chat-app/webapp/src/libs/services/SpeechService.ts:41]: What does invokeToken mean? Are you initializing a new speech service with the token or just grabbing a new token?  If the latter, let's change this to getTokenAsync

File: samples/apps/copilot-chat-app/webapp/src/libs/services/SpeechService.ts [hunk 3]
```diff
@@ -1,23 +1,17 @@
...
-    method?: string;
-}
-
-export class SpeechService {
-    constructor(private readonly serviceUrl: string) {}
-
-    validSpeechKeyAsync = async () => {
-        const response = await this.invokeTokenAsync();
+export class SpeechService extends BaseService {
+    validateSpeechKeyAsync = async (accessToken: string) => {
```
Comment (teresaqhoang) [inline samples/apps/copilot-chat-app/webapp/src/libs/services/SpeechService.ts:13]: What's the point of this method if there's no validation in it? Couldn't you just call invokeTokenAsync?

**Review by teresaqhoang — APPROVED** [review]

**Review by hathind-ms — APPROVED** [review]

**Pull request merged**